ibis/pyspark/compiler.py

```diff
@@ -102,15 +103,25 @@ def compile_sql_query_result(t, expr, scope, timecontext, **kwargs):
 @compiles(ops.Selection)
 def compile_selection(t, expr, scope, timecontext, **kwargs):
     op = expr.op()
-
-    src_table = t.translate(op.table, scope, timecontext)
+    # In selection, there could be multiple children that point to the
+    # same root table. e.g. window with different sizes on a table.
+    # We need to get the 'max' time range that is a superset of every
+    # time context among child nodes, and pass this as context to
+    # source table to get all data within time context loaded.
+    arg_timecontexts = [
+        adjust_context(node.op(), timecontext)
+        for node in op.selections
+        if timecontext
+    ]
+    max_timecontext = union_time_context(arg_timecontexts)
```

Comment:
It's confusing whether this is a "union" or "max" because you used both..

Reply:
Yeah good point, I rename them all to be `combine` for less confusion.